codebase/src/java/disco/org/openlvc/disco/application/EntityStateStore.java

```diff
@@ -69,16 +69,19 @@ protected void receivePdu( EntityStatePdu pdu )
 		// if we are discovering this entity for first time, store in marking indexed store as well
 		if( existing == null )
 		{
-			EntityStatePdu existingMarking = byMarking.put( pdu.getMarking(), pdu );
-			
-			// If there is already an entity against this marking with a different id, then
-			// we assume that it has gone stale in favor of the one that we have just received.
-			//
-			// Note: This check was added for observed behavior when restarting a VR-Forces 
-			// simulation with HLA. The Entities in the scenario are removed and then re-added 
-			// with different EntityIds, however their marking are the same.
-			if( existingMarking != null )
-				byId.remove( existingMarking.getEntityID(), existingMarking );
+			if( !pdu.getMarking().isBlank() )
+			{
+				EntityStatePdu existingMarking = byMarking.put( pdu.getMarking(), pdu );
```

Comment:
This block is now only executed when the marking field is non-blank. It means that we never register entries in the `byMarking` map Entities that have a blank marking.